docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+- `/all` will not require any input from `/[start details]`
+3.	The method `getAllQns()` from package `quizhub.questionlist.QuestionList` will be called to retrieve all questions from the storage list.
+
+`/[qn mode]`
+1.	2 configurations - `/random` and `/normal`
+2.	This prompt is activated after defining `/[quiz mode]` and `/[start details]`
+3.	/random will randomize the list of questions using `java.util.Collections.shuffle` and store it within a temporary array to prevent tempering with the original array in Storage
+4.	`/normal` will not requirwritee any further actions, using the previously generated list as specified by `/[quiz mode]` and `/[start details]`
+
+`/[start details]`
+1.	Define the category of `/module` i.e. `/module` number to call upon the “number” tagged modules when starting the quiz
+2.	This field can be left blank when `/all` is called for `/[quiz mode]` to tell the program to quiz the user on all questions stored within the local storage
+
+Thereafter the quiz is started by calling the method `startQuiz()` in package `quizhub.questionlist.QuestionList`. Within `startQuiz()`, the program iterates through the list of totalQuestions while blocking out the answers. The user can input their answer in the input field which is utilized to match with the actual answer to provide “correct” or “wrong”. Each correct answer will increment correctAnswers variable by 1. The quiz ends when all the questions are displayed and the total number of correctAnswers will be displayed.
+
+### Shuffle Command
+
+#### Brief Description of Shuffle Command
+The Shuffle command allows the user to shuffle quiz questions to a random order PERMANENTLY
+- `shuffle` 
+
+#### Class Structure of Shuffle Command
+![commandShuffleSequence.png](UML%2FCommands%2FcommandShuffleSequence.png)
+
+The "shuffle" command in QuizHub is used to shuffle the order of questions within a question list. 
+This command provides users with the ability to randomize the sequence of questions, which can be useful for 
+creating randomized quizzes or study sessions. 
+
+![ShuffleToStorage-Shuffle_to_Storage_Flow.png](UML%2FCommands%2FShuffleToStorage-Shuffle_to_Storage_Flow.png)
+
+**NOTE:** The randomised sequence will be stored PERMANENTLY in storage. This is unlike /random for command start which 
+creates a temporary array to store the randomised sequence of questions
+
+
+### Markdiff Command - mark difficulty of entry
+
+#### Brief Description of Markdiff Command
+The CommandMarkDifficulty class in the quizhub application is responsible for handling user commands to mark the 
+difficulty of a question. 
+
+The CommandMarkDifficulty class supports the following command syntax:
+- `markdiff [question number] /[question difficulty]` - sets the difficulty of question
+
+#### Class Structure of Markdiff Command
+![commandMarkDiffSequence.png](UML%2FCommands%2FcommandMarkDiffSequence.png)
+
+![commandMarkDiffClass.png](UML%2FCommands%2FcommandMarkDiffClass.png)
```

Comment:
The arrows here are for inheritance, I think it should be something like "->" instead